api/client/client.go

```diff
@@ -0,0 +1,178 @@
+package client
+
+import (
+	"context"
+	"errors"
+
+	"github.com/cosmos/cosmos-sdk/crypto/keyring"
+	logging "github.com/ipfs/go-log/v2"
+	"google.golang.org/grpc"
+
+	"github.com/celestiaorg/celestia-node/blob"
+	blobapi "github.com/celestiaorg/celestia-node/nodebuilder/blob"
+	blobstreamapi "github.com/celestiaorg/celestia-node/nodebuilder/blobstream"
+	fraudapi "github.com/celestiaorg/celestia-node/nodebuilder/fraud"
+	headerapi "github.com/celestiaorg/celestia-node/nodebuilder/header"
+	"github.com/celestiaorg/celestia-node/nodebuilder/p2p"
+	shareapi "github.com/celestiaorg/celestia-node/nodebuilder/share"
+	stateapi "github.com/celestiaorg/celestia-node/nodebuilder/state"
+	"github.com/celestiaorg/celestia-node/state"
+)
+
+var log = logging.Logger("celestia-client")
+
+// TODO:
+// Major limitations:
+// We still carry deps for cosmos-sdk and celestia-app replaces.
+// - can be solved by moving client to a separate package and decoupling all rpcs.
+// Client needs to be reinitialized on every key change due to txClient being stateful and
+// maintaining copy of keys.	- can be reworked by wrapping keyring and updating txClient internal
+// state on key change. Complicated Core accessor requires KeyName to be set and exist in the
+// keyring. Which is not necessary imo. - require refactor of core_accessor/txClient
+// Can't make submit only client, because submition is coupled with reads in blob service
```

Comment:
I don't think that's necessary and if someone really wants to submit without reading they can just use the TxClient

Reply:
Generally agree. There some implementation details that are not yet available in TxClient like fee estiamtion.